### PR TITLE
Update the `super` method within GPT, update README and USAGE file

### PR DESCRIPTION
**Pull request by MarkLee131**
Based on https://github.com/google/oss-fuzz-gen/pull/545, in which we add support for Azure's GPT model, we updated the logic of `_get_tiktoken_encoding` according to @mihaimaruseac 's advice (https://github.com/google/oss-fuzz-gen/pull/545#discussion_r1723731712).

Specifically, we have updated
1. models.py: Updated `_get_tiktoken_encoding` in the GPT and AzureGPT classes, to reuse it in AzureGPT by calling super method.
2. README.md: showed our support for Azure GPT models.
3. USAGE.md: Updated environment settings and model usage for Azure GPT models.

### CODE REVIEW

**Comment by MarkLee131** [comment]
@mihaimaruseac I have finished the update that required in https://github.com/google/oss-fuzz-gen/pull/545. :)

File: USAGE.md [hunk 1]
```diff
@@ -48,13 +48,36 @@ export VERTEX_AI_LOCATIONS=us-west1,us-west4,us-east4,us-central1,northamerica-n
...
+
+Then set them as ENV variables:
+```bash
+export AZURE_OPENAI_API_KEY='<your-azure-api-key>'
+export AZURE_OPENAI_ENDPOINT='<your-azure-endpoint>'
+export AZURE_OPENAI_API_VERSION='<your-azure-api-version>' # default is '2024-02-01'
+```
+
+> Tip: 
+To distinguish between the two ways of accessing OpenAI models, you need to add `-azure` to the model name when **when using OpenAI on Azure**. For example, `gpt-3.5-turbo-azure` will use OpenAI on Azure, while `gpt-3.5-turbo` will use OpenAI on OpenAI.
```
Comment (mihaimaruseac) [inline USAGE.md:79]:
```suggestion
To distinguish between the two ways of accessing OpenAI models, you need to add `-azure` to the model name **when using OpenAI on Azure**. For example, `gpt-3.5-turbo-azure` will use OpenAI on Azure, while `gpt-3.5-turbo` will use OpenAI on OpenAI.
```

**Review by mihaimaruseac — APPROVED** [review]
Thank you!

**Pull request merged**